.github/workflows/windows-artifacts.yml

```diff
@@ -13,69 +13,107 @@ jobs:
   windows-artifacts:
     permissions:
       contents: write # to fetch code and upload artifacts
-    # see https://ariya.io/2020/07/on-github-actions-with-msys2
-    runs-on: windows-latest
-    # see https://github.com/msys2/setup-msys2
+    # For msys2, see https://ariya.io/2020/07/on-github-actions-with-msys2
+    runs-on: ${{ matrix.build_type == 'cmake' && 'windows-11-arm' || 'windows-latest' }}
     strategy:
+      # For msys2, see https://github.com/msys2/setup-msys2
       matrix:
         include:
-          - { msystem: mingw64, env: x86_64, ziparch: win64 }
-          - { msystem: mingw32, env: i686, ziparch: win32 }
+          - { msystem: mingw64, env: x86_64, ziparch: win64, build_type: msys2 }
+          - { msystem: mingw32, env: i686, ziparch: win32, build_type: msys2 }
+          - { msystem: null, env: arm64, ziparch: win-arm64, build_type: cmake }
+    
     defaults:
       run:
-        shell: msys2 {0}
+        shell: ${{ matrix.build_type == 'cmake' && 'pwsh' || 'msys2 {0}' }}
```

Comment:
There are more idiomatic ways to state the same thing, with less long-context to maintain, 
for example:

```YAML
- name: Build (MSYS2)
  if: matrix.build_type == 'msys2'
  shell: msys2 {0}
  run: |
    # msys2 commands here

- name: Build (CMake/ARM64)
  if: matrix.build_type == 'cmake'
  shell: pwsh
  run: |
    # PowerShell commands here
 ```

I have not yet understood why Powershell `pwsh` is necessary here. `msys2` also provides `cmake`
 
On the other hand, if both code paths have a lot of differences, maybe it would be better to separate them completely into different actions.